File: 프로그래머스/lv0/120890. 가까운 수/가까운 수.js
function solution(array, n) {
  // array 의 요소 중 n 과 가까운 수
  let result = array[0];

  for (let i = 1; i < array.length; i++) {
    const temp = array[i];
    const currentDifference = Math.abs(n - temp);
    const resultDifference = Math.abs(n - result);

    if (
      currentDifference < resultDifference ||
      (currentDifference === resultDifference && temp < result)
    ) {
      result = temp;
    }
  }

  return result;

}

File: 프로그래머스/lv0/120890. 가까운 수/가까운 수.java
import java.util.Arrays;
class Solution {
    public int solution(int[] array, int n) {
        return Arrays.stream( array )
					 .reduce( array[ 0 ], ( acc, curr ) -> {
						 int accValue = Math.abs( acc - n );
						 int currValue = Math.abs( curr - n );

						 if ( accValue > currValue || accValue == currValue && acc > curr )
							 return curr;

						 return acc;
					 } );
    }
}